chia/full_node/mempool_manager.py

```diff
@@ -184,16 +183,17 @@ async def add_spendbundle(
         if self.peak is None:
             return None, MempoolInclusionStatus.FAILED, Err.MEMPOOL_NOT_INITIALIZED
 
-        npc_list = cost_result.npc_list
-        cost = cost_result.cost
+        npc_list = npc_result.npc_list
+        program = simple_solution_generator(new_spend).program
+        cost = calculate_cost_of_program(program, npc_result, self.constants.COST_PER_BYTE)
```

Comment:
This snippet looks suspicious to me. `simple_solution_generator()` creates the program, we pass the program to `calculate_cost_of_program()`. but with this PR, `calculate_cost_of_program()` doesn't run the CLVM code. So where do the `npc_results` come from? do they come from having run a different program?

Reply:
We prevalidate the spendbundle beforehand

Reply:
I think (in the future) we should not pass in `program` or `COST_PER_BYTE`, but instead record that cost in the `npc_result` where we record the cost of running the program.

Reply:
yeah that makes sense